recipes/ncbi-cxx-toolkit-public/26.0/conanfile.py

```diff
@@ -0,0 +1,1532 @@
+import os
+import shutil
+import subprocess
+import tempfile
+import sys
+import stat
+
+def _remove_readonly(fn, path, excinfo):
+    try:
+        os.chmod(path, stat.S_IWRITE)
+        fn(path)
+    except Exception as exc:
+        print("Skipped: ", path, " because ", exc)
+
+class NcbiCxxToolkit(ConanFile):
+    name = "ncbi-cxx-toolkit-public"
+    license = "CC0-1.0"
+    homepage = "https://ncbi.github.io/cxx-toolkit"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "NCBI C++ Toolkit"
+    topics = ("ncbi", "biotechnology", "bioinformatics", "genbank", "gene",
+              "genome", "genetic", "sequence", "alignment", "blast",
+              "biological", "toolkit", "c++")
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    tk_tmp_tree = ""
+    tk_src_tree = ""
+    all_NCBI_requires = {}
+
+    options = {
+        "shared":     [True, False],
+        "fPIC":       [True, False],
+        "projects":   "ANY",
+        "targets":    "ANY"
+    }
+    default_options = {
+        "shared":     False,
+        "fPIC":       True,
+        "projects":   "",
+        "targets":    "",
+    }
+    NCBI_to_Conan_requires = {
+        "BerkeleyDB":   "libdb/5.3.28@#355aa134e9ee1bda60fa224f8b54c913",
+        "BZ2":          "bzip2/1.0.8@#493645787acdd3909523a6a2773b1b36",
+        "CASSANDRA":    "cassandra-cpp-driver/2.15.3@#27c46dd9dfbb2a3afef30aa2e914f00a",
+        "GIF":          "giflib/5.2.1@#75a273d91a6c63bc0d8a93799ce19acc",
+        "JPEG":         "libjpeg/9d@#956b5e73632aac77088571f574d766c0",
+        "LMDB":         "lmdb/0.9.29@#3bdb51ff4c3649bd865ee3857f0f2e07",
+        "LZO":          "lzo/2.10@#76a80652793ae478a12f6f9a8eb17a89",
+        "MySQL":        "libmysqlclient/8.0.25@#bbe62517865849c442c032cc7aecfd57",
+        "NGHTTP2":      "libnghttp2/1.45.1@#cef166f78349143079e1c667d823b25a",
+        "PCRE":         "pcre/8.45@#b6bae758d36ca2050bb8527ea408eb75",
+        "PNG":          "libpng/1.6.37@#8b23717a9884a2b06884c599072955f4",
+        "SQLITE3":      "sqlite3/3.36.0@#567836d4f4b7b58a4e834bf98288d609",
+        "TIFF":         "libtiff/4.3.0@#28ac662980826ae8ffe8d6236cdd9735",
+        "XML":          "libxml2/2.9.12@#9817dd585ffc6de1479da6d5bcf01fc0",
+        "XSLT":         "libxslt/1.1.34@#47946f5c7abe03d18179b55be67bbabe",
+        "UV":           "libuv/1.42.0@#e1801aae9570062012d94db20338451b",
+        "Z":            "zlib/1.2.11@#683857dbd5377d65f26795d4023858f9"
+    }
+
+#----------------------------------------------------------------------------
+    def __del__(self):
+        if os.path.isdir(self.tk_tmp_tree):
+            print("Just a moment...")
+            self._remove_tmp_tree(self.tk_tmp_tree)
+
+    def _remove_tmp_tree(self, path):
+        if os.path.isdir(path):
+            os.chdir(os.path.join(self.tk_tmp_tree, ".."))
+            try:
+                shutil.rmtree(path, onerror=_remove_readonly)
```

Comment:
Does https://docs.conan.io/en/latest/reference/tools.html#tools-mkdir-tools-rmdir support this?

Reply:
maybe, so what?